processor/src/main/java/org/mapstruct/ap/internal/model/common/Parameter.java

```diff
@@ -107,4 +101,27 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    /**
+     * @param classTypeOf the type representing {@code Class<X>} for the target type {@code X}
+     * @return a {@link Parameter} usable for parameter bindings
+     */
+    public static Parameter forTargetTypeBinding(Type classTypeOf) {
+        return new Parameter( null, classTypeOf, false, true );
```

Comment:
Yea.. one more reason to create a new class for that. What's better, `ParameterBinding` or `MethodReferenceArgument`?